hie-plugin-api/Haskell/Ide/Engine/PluginsIdeMonads.hs

```diff
@@ -261,6 +263,7 @@ data FormattingType = FormatText
 -- and file uri, does not suffice.
 type FormattingProvider = T.Text -- ^ Text to format
         -> Uri -- ^ Uri of the file being formatted
+        -> Maybe Cradle -- ^ HIE.Bios Cradle containing the file being formatted
```

Comment:
I would rather the cradle be accessed via a method of `IdeM` rather than passing it, since accessing the cradle is also probably useful in other contexts, and means we don't have to extract it out and pass it along inside the reactor